tests/how_to_test_systems.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+
+use bevy::prelude::*;
+
+struct Enemy {
+    hit_points: u32,
+}
+
+struct CharacterTemplate {
```

Comment:
I'm not sure the CharacterTemplate adds much value here (and feels like an "un-bevy" thing to do). Imo we should encourage things like constructors or setting public fields, unless we are creating a _specific_ Enemy, in which case it should be a function or a Bundle.

I think the point of adding this was to illustrate how to setup and use "test resources", but I think as-is it risks encouraging anti-patterns for entity construction.

Reply:
changed the test to use the much more interesting-to-show-how-to-test resource `Input<KeyCode>`